### DISCUSSION TITLE
Find command not working with lazyvim

### DISCUSSION THREAD

**Discussion by AtaraxiaZ** [LazyVim/LazyVim#1384] [Q&A] [answered]
Hey, I find a very weird problem and I can't work it out since I am a newbee. 

The problem is: I add my project path to the vim path using `vim.opt.path:append("/my/project/path/**")`, and I can autocomplete with tab when I type `:find my_file`. But when I click enter, it shows "E345: can't find file my_file in the path". 

The weird thing is I have autocomplete of the file name which will not be there if I delete the `vim.opt.path:append`, and I use `set path` to check that my project path is indeed there. I tried neovim with no plugins mode and all went well. 

So I guess it maybe one of the plugin in lazyvim cause this problem, and I use a brand new lazyvim starter to test, still the same problem, so it can't be one of my plugins added by myself caused the problem. But lazyvim have so many plugins so I can't figure it out which one cause the issue. Hope anyone could help me , Thanks!

**Comment by dpetka2001** [comment]
Try using an absolute path for the project you want to declare. Something like `/home/user/my_project/**`.  I tried with `~/my_project/**` and got the same error you got. Maybe it can't do variable expansion?

**Reply by AtaraxiaZ** [reply]
Great!  I sovled it by using /home/user/. Thanks for your answer!

**Comment by abeldekat** [comment]
I tried it as well, and E345 only occurs when I am using the "~" in the path. Moreover, it also happens when using `nvim --noplugin`.

You can use: `lua vim.opt.path:append(vim.env.HOME .. "/projects/some-project/**")`

`:h path`:
> - Environment variables are expanded |:set_env|.

**Reply by dpetka2001** [reply, marked as answer]
To add to this, from `:h expand()`, I quote
> Also note that expanding "cfile" and others only returns the
		referenced file name without further expansion.  If "cfile"
		is `~/.cshrc` you need to do another `expand()` to have the
		"~/" expanded into the path of the home directory:
		`:echo expand(expand("cfile"))`

So, you either have to use Environment variables like @abeldekat already mentioned or expand the filename 
yourself if you want it to include `~` character in it. Something like 
`lua vim.opt.path:append(vim.fn.expand("~").."/my_project/**")`, which is basically the same as `vim.env.HOME`.

**Reply by jyuan0** [reply]
To add another tidbit: yet another option is `os.getenv("HOME")` - I tend to forget that the Lua standard libraries are available :)
